x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlActionIT.java

```diff
@@ -0,0 +1,95 @@
+import org.elasticsearch.action.bulk.BulkRequestBuilder;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.search.SearchHit;
+import org.junit.BeforeClass;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static org.elasticsearch.test.StreamsUtils.copyToStringFromClasspath;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+
+public class EqlActionIT extends AbstractEqlIntegTestCase {
+
+    @BeforeClass
+    public static void checkForSnapshot() {
+        assumeTrue("Only works on snapshot builds for now", Build.CURRENT.isSnapshot());
+    }
+
+    public void testEqlSearchAction() throws Exception {
+        final String indexPrefix = "endgame";
+        final String testIndexName = indexPrefix + "-1.4.0";
+        final String testIndexPattern = indexPrefix + "-*";
+
+        String endgame = copyToStringFromClasspath("/endgame.json");
+        endgame = endgame.replace("[index_pattern_placeholder]", testIndexPattern);
+
+        assertAcked(client().admin().indices().preparePutTemplate(testIndexName)
+            .setSource(endgame.getBytes(StandardCharsets.UTF_8), XContentType.JSON).get());
+
+        // Insert test data
+        InputStream is = EqlActionIT.class.getResourceAsStream("/endgame.dat");
+        BulkRequestBuilder bulkBuilder = client().prepareBulk();
+        try (BufferedReader reader = new BufferedReader(
+            new InputStreamReader(is, StandardCharsets.UTF_8))) {
+
+            String line;
+            while ((line = reader.readLine()) != null) {
+                bulkBuilder.add(new IndexRequest(testIndexName).source(line.trim(), XContentType.JSON));
+            }
+            BulkResponse response = bulkBuilder.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).get();
+            assertThat(response.hasFailures() ? response.buildFailureMessage() : "", response.hasFailures(), equalTo(false));
+        }
+
+        ensureYellow(testIndexName);
+
+        // Load EQL validation specs
+        List<EqlSpec> specs = EqlSpecLoader.load("/test_queries.toml", true);
+        List<EqlSpec> unsupportedSpecs = EqlSpecLoader.load("/test_queries_unsupported.toml", false);
+
+        // Validate only currently supported specs
+        for (EqlSpec spec : specs) {
+            boolean supported = true;
+            // Check if spec is supported, simple iteration, cause the list is short.
+            for (EqlSpec unSpec : unsupportedSpecs) {
+                if (spec.query.equals(unSpec.query)) {
+                    supported = false;
+                    break;
+                }
+            }
+
+            if (supported) {
```

Comment:
I think this test should use `@ParametersFactory` so each example becomes a test that we can address individually.